backends/ref/ceed-ref-restriction.c

```diff
@@ -272,34 +290,66 @@ static int CeedElemRestrictionDestroy_Ref(CeedElemRestriction r) {
 // ElemRestriction Create
 //------------------------------------------------------------------------------
 int CeedElemRestrictionCreate_Ref(CeedMemType mtype, CeedCopyMode cmode,
-                                  const CeedInt *indices, CeedElemRestriction r) {
+                                  const CeedInt *offsets,
+                                  CeedElemRestriction r) {
   int ierr;
   CeedElemRestriction_Ref *impl;
-  CeedInt elemsize, nelem;
+  CeedInt nelem, elemsize, numblk, blksize, ncomp, compstride;
   ierr = CeedElemRestrictionGetNumElements(r, &nelem); CeedChk(ierr);
   ierr = CeedElemRestrictionGetElementSize(r, &elemsize); CeedChk(ierr);
+  ierr = CeedElemRestrictionGetNumBlocks(r, &numblk); CeedChk(ierr);
+  ierr = CeedElemRestrictionGetBlockSize(r, &blksize); CeedChk(ierr);
+  ierr = CeedElemRestrictionGetNumComponents(r, &ncomp); CeedChk(ierr);
+  ierr = CeedElemRestrictionGetCompStride(r, &compstride); CeedChk(ierr);
   Ceed ceed;
   ierr = CeedElemRestrictionGetCeed(r, &ceed); CeedChk(ierr);
 
   if (mtype != CEED_MEM_HOST)
     // LCOV_EXCL_START
     return CeedError(ceed, 1, "Only MemType = HOST supported");
   // LCOV_EXCL_STOP
-  ierr = CeedCalloc(1,&impl); CeedChk(ierr);
+  ierr = CeedCalloc(1, &impl); CeedChk(ierr);
+
+  // Check indices for ref or memcheck backends
+  if (offsets) {
+    Ceed parentCeed = ceed, currCeed = NULL;
+    while (parentCeed != currCeed) {
+      currCeed = parentCeed;
+      ierr = CeedGetParent(currCeed, &parentCeed); CeedChk(ierr);
+    }
+    const char *resource;
+    ierr = CeedGetResource(parentCeed, &resource); CeedChk(ierr);
+    if (!strcmp(resource, "/cpu/self/ref/serial")
+        || !strcmp(resource, "/cpu/self/ref/blocked")
+        || !strcmp(resource, "/cpu/self/memcheck/serial")
+        || !strcmp(resource, "/cpu/self/memcheck/blocked")) {
+      CeedInt lsize;
+      ierr = CeedElemRestrictionGetLVectorSize(r, &lsize); CeedChk(ierr);
+
+      for (CeedInt i = 0; i < nelem*elemsize; i++)
+        if (offsets[i] > lsize - 1)
```

Comment:
Is this stricter variant correct based on your current assumptions?
```suggestion
        if (offsets[i] < 0 || lsize <= offsets[i] + (ncomp - 1) * compstride)
```